SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -293,26 +297,26 @@ void FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
 		}
 
 		// Timeout detection.
-		if (bStartingDeployment && Output.Contains(TEXT("startup completed")))
+		if (!bLocalDeploymentRunning && Output.Contains(TEXT("startup completed")))
 		{
-			bStartingDeployment = false;
+			bLocalDeploymentRunning = true;
 		}
 	});
 
 	RuntimeProcess->Launch();
 
-	while (bStartingDeployment && RuntimeProcess->Update())
+	// Wait for runtime to start or timeout
+	while (!bLocalDeploymentRunning && RuntimeProcess->Update() && RuntimeProcess->GetDuration().GetTotalSeconds() <= RuntimeTimeout)
 	{
-		if (RuntimeProcess->GetDuration().GetTotalSeconds() > RuntimeTimeout)
-		{
-			UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Timed out waiting for the Runtime to start."));
-			bStartingDeployment = false;
-			break;
-		}
+		std::this_thread::sleep_for(std::chrono::microseconds(100));
```

Comment:
I think this should be replaceable with an unreal version?
Maybe `FPlatformProcess::Sleep`? But I haven't looked into it too much.